comictaggerlib/taggerwindow.py

```diff
@@ -2151,14 +2169,25 @@ def load_archive(self, comic_archive: ComicArchive) -> None:
 
         self.config[0].internal__last_opened_folder = os.path.abspath(os.path.split(comic_archive.path)[0])
         self.comic_archive = comic_archive
+
+        self.metadata, success = self.overlay_ca_read_style(self.load_data_styles, self.comic_archive)
+        if not success:
+            self.exception(f"Failed to load metadata for {self.comic_archive.path}, see log for details\n\n")
+
+        self.update_ui_for_archive()
+
+    def overlay_ca_read_style(self, load_data_styles: list[str], ca: ComicArchive) -> tuple[GenericMetadata, bool]:
+        md = GenericMetadata()
+        success = True
         try:
-            self.metadata = self.comic_archive.read_metadata(self.load_data_style)
+            for style in reversed(load_data_styles):
```

Comment:
Instead of reversing it here reverse it when setting/getting from the Qt widget.